executor/builder.go

```diff
@@ -1923,3 +1876,28 @@ func buildKvRangesForIndexJoin(sc *stmtctx.StatementContext, tableID, indexID in
 	})
 	return kvRanges, nil
 }
+
+func (b *executorBuilder) buildWindow(v *plannercore.PhysicalWindow) *WindowExec {
+	childExec := b.build(v.Children()[0])
+	if b.err != nil {
+		return nil
+	}
+	base := newBaseExecutor(b.ctx, v.Schema(), v.ExplainID(), childExec)
+	var groupByItems []expression.Expression
+	for _, item := range v.PartitionBy {
+		groupByItems = append(groupByItems, item.Col)
+	}
+	aggDesc := aggregation.NewAggFuncDesc(b.ctx, v.WindowFuncDesc.Name, v.WindowFuncDesc.Args, false)
+	resultColIdx := len(v.Schema().Columns) - 1
+	agg := aggfuncs.Build(b.ctx, aggDesc, resultColIdx)
+	if agg == nil {
+		b.err = errors.Trace(errors.New("window evaluator only support aggregation functions without frame now"))
+		return nil
+	}
+	e := &WindowExec{baseExecutor: base,
+		windowFunc:    agg,
+		partialResult: agg.AllocPartialResult(),
+		groupChecker:  newGroupChecker(b.ctx.GetSessionVars().StmtCtx, groupByItems),
```

Comment:
we should use `partition by` columns to separate partitions, and use `order by` columns to further separate groups inside a partition, for example, in MySQL:
```sql
MySQL(root@localhost:test) > select a, b, sum(b) over w from t window w as (partition by a order by b);
+------+------+---------------+
| a    | b    | sum(b) over w |
+------+------+---------------+
|    1 |    1 |             2 |
|    1 |    1 |             2 |
|    1 |    2 |             4 |
|    2 |    1 |             1 |
+------+------+---------------+
4 rows in set (0.00 sec)
```

The first three rows belongs to the same partition (a = 1), and this partition contains two groups, i.e. b = 1 and b = 2, window function result for each group shares the same result

Reply:
It is a problem that can only be solved after we support frame clause. According to https://dev.mysql.com/doc/refman/8.0/en/window-functions-frames.html, when a window definition has order by but without frame clause, the default frame includes rows from the partition start through the current row, including all peers of the current row (rows equal to the current row according to the ORDER BY clause).

Reply:
got it.